### PR TITLE
AGP8 compatibility

### PR DESCRIPTION
**Pull request by deckerst**
Main change is adding namespace for AGP8 compatibility.
Rest is upgrades so that the example can compile on a modern setup with recent Flutter/Gradle.

### CODE REVIEW

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -55,7 +55,7 @@ android {
 
     defaultConfig {
         applicationId "com.nimroddayan.flutternsd.example"
-        minSdkVersion 23
+        minSdkVersion 21
```
Comment (Nimrodda) [inline example/android/app/build.gradle]: Please keep the existing minimum

File: android/build.gradle [hunk 2]
```diff
@@ -19,30 +19,35 @@ group 'com.nimroddayan.flutternsd'
...
+    ext.kotlin_version = '1.7.10'
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.1'
+        classpath 'com.android.tools.build:gradle:7.3.1'
```
Comment (Nimrodda) [inline android/build.gradle:29]: PR title says AGP 8 but I see 7 here :)

**Review by Nimrodda — CHANGES_REQUESTED** [review]
Thanks for the PR. I left some comments.
Also have you verified the the example app works?

File: android/build.gradle (hunk 2 above)
Comment (deckerst) [reply android/build.gradle:29]: AGP 8 compatibility is for __consumer__ projects using AGP 8.
Here changing to 7 was not necessary for AGP 8 compatibility, but just part of general upgrades from ancient versions used in this repo.

File: example/android/app/build.gradle (hunk 1 above)
Comment (deckerst) [reply example/android/app/build.gradle]: Ok. By the way, is there a reason for this minimum?

**Comment by deckerst** [comment]
> Also have you verified the the example app works?

Yes. And this fix for AGP8 has been used in many repos already.

**Comment by Nimrodda** [comment]
checks failed `Because flutter_nsd_example depends on cupertino_icons >=1.0.6 which requires SDK version >=2.19.0 <4.0.0, version solving failed.`. Let's address that before I merge this.

**Comment by deckerst** [comment]
Interesting. In my environment I use a recent Dart SDK which requires null safety, so I needed to bump `cupertino_icons` version to make it build. But your GitHub workflow uses an old Dart SDK (< 2.19.0). That Dart SDK version depends on the Flutter version you specified in `env.FLUTTER_VERSION` which is setup at the repo level. So my understanding is that you can bump that `FLUTTER_VERSION`. Otherwise, feel free to reject this PR.

**Comment by Nimrodda** [comment]
> Interesting. In my environment I use a recent Dart SDK which requires null safety, so I needed to bump `cupertino_icons` version to make it build. But your GitHub workflow uses an old Dart SDK (< 2.19.0). That Dart SDK version depends on the Flutter version you specified in `env.FLUTTER_VERSION` which is setup at the repo level. So my understanding is that you can bump that `FLUTTER_VERSION`. Otherwise, feel free to reject this PR.

I use [FVM](https://github.com/Nimrodda/flutter_nsd/blob/master/.fvm/fvm_config.json) for Flutter versioning. Could you please update it and we see if it fixes the issue?

**Pull request merged**